lib/allocation.c

```diff
@@ -210,14 +211,17 @@ get_unused_cos(const unsigned id,
 
         /* Create a list of COS used on socket/L2 cluster */
         for (i = 0; i < m_cpu->num_cores; i++) {
-
                 if (l2_req) {
                         /* L2 requested so looking in L2 cluster scope */
                         if (m_cpu->cores[i].l2_id != id)
                                 continue;
+		} else if (mba_req) {
```

Comment:
You assume that MBA contains a subset od cores from CAT group

Reply:
I am not very clear about this comment.  Please take a look at the new code. thanks

Reply:
On Intel CPU's we set CAT and MBA settings on package level. As I understand on AMD cpu MBA/CAT setting are set per mba/cat_id.

Please consider the following configuration:

CORE | MBA_ID | CATL3_ID
------ | --------- | -------
0 | 1 | 1
1 | 2 | 1

When MBA is requested for core 0 it might not detect that L3 CAT COS is already assigned to core 1.

Reply:
Normally for one core,  both MBA ID and CAT ID is same. Yes. this is a hypothetical case.  We can change the allocation in hw_alloc_assign to set on core basis for each technology separately if that is what you are thinking about. That way it will go over each core and detect the cat or mba id and set the allocation. If you think we can clarify things over the call quickly, we can have a quick call. Let me know.

Reply:
Ok, We can assume that both MBA ID and CAT ID are the same. Just put a comment about that.

Reply:
ok. Added comment in get_unused_cos. Thanks